src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -0,0 +1,731 @@
+
+    FD_LOG_DEBUG(( "start genesis accounts - count: %lu", genesis_block.accounts_len));
+
+    for( ulong i=0; i < genesis_block.accounts_len; i++ ) {
+      fd_pubkey_account_pair_t * a = &genesis_block.accounts[i];
+
+      FD_BORROWED_ACCOUNT_DECL(rec);
+
+      int err = fd_acc_mgr_modify(
+        ctx->slot_ctx->acc_mgr,
+        ctx->slot_ctx->funk_txn,
+        &a->key,
+        /* do_create */ 1,
+        a->account.data_len,
+        rec);
+      if( FD_UNLIKELY( err ) )
+        FD_LOG_ERR(( "fd_acc_mgr_modify failed (%d)", err ));
+
+      rec->meta->dlen            = a->account.data_len;
+      rec->meta->info.lamports   = a->account.lamports;
+      rec->meta->info.rent_epoch = a->account.rent_epoch;
+      rec->meta->info.executable = a->account.executable;
+      memcpy( rec->meta->info.owner, a->account.owner.key, 32UL );
+      if( a->account.data_len )
+        memcpy( rec->data, a->account.data, a->account.data_len );
+    }
+
+    FD_LOG_DEBUG(( "end genesis accounts"));
+
+    FD_LOG_DEBUG(( "native instruction processors - count: %lu", genesis_block.native_instruction_processors_len));
+
+    for( ulong i=0; i < genesis_block.native_instruction_processors_len; i++ ) {
+      fd_string_pubkey_pair_t * a = &genesis_block.native_instruction_processors[i];
+      fd_write_builtin_bogus_account( ctx->slot_ctx, a->pubkey.uc, a->string, strlen(a->string) );
+    }
+
+    /* sort and update bank hash */
+    int result = fd_update_hash_bank( ctx->slot_ctx, capture_ctx, &ctx->slot_ctx->slot_bank.banks_hash, ctx->slot_ctx->signature_cnt );
+    if (result != FD_EXECUTOR_INSTR_SUCCESS) {
+      FD_LOG_ERR(("Failed to update bank hash with error=%d", result));
+    }
+
+    ctx->slot_ctx->slot_bank.slot = 0UL;
+  }
+  FD_TEST( FD_RUNTIME_EXECUTE_SUCCESS == fd_runtime_save_epoch_bank( ctx->slot_ctx ) );
+
+  FD_TEST( FD_RUNTIME_EXECUTE_SUCCESS == fd_runtime_save_slot_bank( ctx->slot_ctx ) );
+
+  fd_bincode_destroy_ctx_t ctx2 = { .valloc = ctx->slot_ctx->valloc };
+  fd_genesis_solana_destroy(&genesis_block, &ctx2);
+
+  // if( capture_ctx )  {
+  //   fd_solcap_writer_fini( capture_ctx->capture );
+  // }
+}
+
+static void
+read_snapshot( void * _ctx, char const * snapshotfile, char const * incremental ) {
+  fd_replay_tile_ctx_t * ctx = (fd_replay_tile_ctx_t *)_ctx;
+
+  const char * snapshot = snapshotfile;
+
+  fd_valloc_t valloc = ctx->slot_ctx->valloc;
+  ctx->slot_ctx->valloc = inner_valloc;
+
+  fd_snapshot_load(snapshot, ctx->slot_ctx, false, false, FD_SNAPSHOT_TYPE_FULL );
+  if ( strlen(incremental) > 0 ) {
+    ctx->epoch_ctx = fd_exec_epoch_ctx_join( fd_exec_epoch_ctx_new( ctx->epoch_ctx_mem, 2000000UL ) );
+
+    fd_snapshot_load(incremental, ctx->slot_ctx, false, false, FD_SNAPSHOT_TYPE_INCREMENTAL );
+  }
+
+  fd_blockstore_snapshot_insert( ctx->slot_ctx->blockstore, &ctx->slot_ctx->slot_bank );
+  ctx->slot_ctx->valloc = valloc;
+}
+
+static void
+during_housekeeping( void * _ctx ) {
```

Comment:
This should handle sync updates for other outputs. Eg poh_out_sync

Reply:
Done